### ISSUE TITLE
e2e test fails with  TypeError: cannot unpack non-iterable NoneType object

### ISSUE THREAD

**Issue by marrobi** [microsoft/AzureTRE#1145]
**Describe the bug**
End to end tests faile - https://github.com/microsoft/AzureTRE/runs/4880000057?check_suite_focus=true with:

```
test_workspace_service_creation.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
helpers.py:116: in disable_and_delete_workspace
    await wait_for(delete_done, client, workspace_id, owner_auth_headers, strings.RESOURCE_STATUS_DELETING_FAILED)
helpers.py:88: in wait_for
    done, done_state, message = await func(client, workspace_id, headers)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

client = <httpx.AsyncClient object at 0x7f5e6e00d100>
workspace_id = '6cddda47-568b-4e8b-9af7-bdf6dd6d5405'
headers = {'Authorization': 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1yNS1BVWliZkJpaTdOZDFqQmViYXhib1hXMCIsImtpZCI6...1cv-KB7xIcly-vLkYp3v3gzL7mPYZSiwqo2xy6-r_A32L2g2QE5RuVtlx2H1UKWTJwULyADcogORksimENhope9ojjZJ9PfM1CwUnozKwIFZHauf7VSsw'}

    async def delete_done(client, workspace_id, headers):
        delete_terminal_states = [strings.RESOURCE_STATUS_DELETED, strings.RESOURCE_STATUS_DELETING_FAILED]
>       deployment_status, message = await check_deployment(client, workspace_id, headers)
E       TypeError: cannot unpack non-iterable NoneType object

helpers.py:80: TypeError
```

**Comment by msebragge** [comment]
I think the response code here (helpers.py line 45) is something else than 200 or 404 and thus the error
```python
async def check_deployment(client, workspace_id, headers) -> (str, str):
    response = await client.get(
        f"https://{config.TRE_ID}.{config.RESOURCE_LOCATION}.cloudapp.azure.com{strings.API_WORKSPACES}/{workspace_id}",
        headers=headers)
    if response.status_code == 200:
        deployment_status = response.json()["workspace"]["deployment"]["status"]
        message = response.json()["workspace"]["deployment"]["message"]
        return deployment_status, message
    elif response.status_code == 404:
        # Seems like the resource got deleted
        return strings.RESOURCE_STATUS_DELETED, "Workspace was deleted"
```

**Comment by damoodamoo** [comment]
Hopefully #1153 should fix this as the test has been rewritten...

**Comment by damoodamoo** [comment]
fixed via PR #1153
